apply_extra.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/bash
+
+unappimage *.AppImage
+mv squashfs-root/* .
+rm -rf squashfs-root *.AppImage
+# asar is a binary archive format, so the replacement string *must* be the same length
+# as the original.
+patch-resources resources/app.asar
+# sed -i -f - resources/app.asar <<'EOF'
+# EOF
+rm AppRun
+mkdir -p export/share/applications
+sed '/X-AppImage/d;s/Exec=AppRun/Exec=start-unityhub/;s/Icon=unityhub/Icon=com.unity.UnityHub/' \
```

Comment:
Producing the .desktop file at install like this means that appstream-compose at build time will not be able to pull the metadata out of the .desktop file into the appdata, so you will need to duplicate icon, category, etc info in the appdata. It's a bit safer / more consistent to just copy the icon and .desktop file into this repo so they are always present in the repo. (Provided the description makes clear this a third-party re-packaging of the Unity Hub and not supported by Unity Technologies, it's OK for us to use the icon/etc to refer to their product.)

Reply:
(also I believe the FlatHub-ists prefer desktop-file-edit than sed, but shipping the file as data would alleviate this concern anyway)

Reply:
Maybe this isn't a real concern at all... I guess we have to maintain our own appdata file which can easily contain the icon and categories anyway...